dev/TabView/APITests/TabViewTests.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using MUXControlsTestApp.Utilities;
+using System;
+using System.Threading;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Markup;
+using Windows.UI.Xaml.Media;
+using Common;
+using Microsoft.UI.Xaml.Controls;
+using System.Collections.Generic;
+
+#if USING_TAEF
+using WEX.TestExecution;
+using WEX.TestExecution.Markup;
+using WEX.Logging.Interop;
+#else
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.VisualStudio.TestTools.UnitTesting.Logging;
+#endif
+
+using Symbol = Windows.UI.Xaml.Controls.Symbol;
+
+namespace Windows.UI.Xaml.Tests.MUXControls.ApiTests
+{
+
+    [TestClass]
+    public class TabViewTests : ApiTestBase
+    {
+
+        [TestMethod]
+        public void VerifyCompactTabWidthVisualStates()
+        {
+            TabView tabView = null;
+            RunOnUIThread.Execute(() =>
+            {
+                tabView = new TabView();
+                Content = tabView;
+
+                tabView.TabItems.Add(CreateTabViewItem("Item 0", Symbol.Add));
+                tabView.TabItems.Add(CreateTabViewItem("Item 1", Symbol.AddFriend));
+                tabView.TabItems.Add(CreateTabViewItem("Item 2"));
+
+                tabView.SelectedIndex = 0;
+                tabView.SelectedItem = tabView.TabItems[0];
+                (tabView.SelectedItem as TabViewItem).IsSelected = true;
+                Verify.AreEqual("Item 0", (tabView.SelectedItem as TabViewItem).Header);
+                tabView.TabWidthMode = TabViewWidthMode.Compact;
+                Content.UpdateLayout();
+            });
+
+            IdleSynchronizer.Wait();
```

Comment:
With the use of ApiTestBase and Content.UpdateLayout() you don't need to call IdleSynchronizer after you call UpdateLayout().  I would replace all these calls with calls to UpdateLayout.

Reply:
Using Content.UpdateLayout() did not work as we/I expected, and the Visual tree was not updated, resulting in test failure. Do you have any idea why that could be?
I have reverted the test in the mean time.

Reply:
UpdateLayout is a synchronous call that will return right after layout pass completes. If there are any async items that are kicked out of that then those will not have completed (animations for example). WaitForIdle will make sure that everything is settled down before it returns.

Reply:
The question is, whether there are any async calls in there or not. Is changing visual states something that could be async?

Reply:
Yes. There can be animations/storyboards that are run as part of the visual state changes.